content-updates-round3 (#3413)
twilio-labs/paste: packages/paste-website/src/pages/components/tooltip/index.mdx

### Accessibility

A tooltip must be placed on **only a natively focusable HTML element**. Good candidates include a [Button](/components/button) or a [Anchor](/components/anchor) if the tip also links to a help article.

> **Do not** place tooltips on non-focusable elements, like an icon or disabled buttons.

Author: Updated to read:

A tooltip must be placed on **only a natively focusable HTML element**, like an enabled [Button](/components/button) or an [Anchor](/components/anchor). **Do not** place tooltips on non-focusable elements, like an icon, or disabled components, like a button. Keyboard users cannot focus on disabled components.

If your tooltip wraps a natively focusable HTML element that includes only an icon, make sure you set the prop `decorative={false}`, and
give the icon a title. The title of the icon should be the accessible name for the button action, like "Delete phone number". The tooltip provides the additional context, like "You can delete phone numbers every 7 days".
This ensures the icon and tooltip are accessible to screen readers. Refer to the [focusable element example](#focusable-element) for implementation.

> When the user focuses an element with a tooltip, their focus stays on the element. Since focus never goes inside the tooltip, avoid using additional components, like an [Anchor](/components/anchor), within the Tooltip. 

Author: done.

  {StateHookTabsExample}
</LivePreview>

## Composition notes

Tooltips should contain only information that is useful additional context or guidance for the user. Furthermore, Tooltips are easily overlooked and disappear when not focused on, so consider whether the information may be better served by another component.

> Never put information essential to task completion in a Tooltip. Use [Help Text](/components/help-text), which is persistent and more obvious, for essential information.
> 
> Do not use a Tooltip on a disabled [Button](/components/button) to explain why the button is unavailable. The disabled button isn't focusable. 
> 
> Similarly, do not including an [Anchor](/components/anchor) in Tooltip content because an Anchor isn't focusable. If you need to include a link, use a [Popover](/components/popover) instead.

Reviewer: ```suggestion
Never put information essential to task completion in a Tooltip. Use [Help Text](/components/help-text), which is persistent and more obvious, for essential information. Similarly, do not include an [Anchor](/components/anchor) in Tooltip content because keyboard users can't reach Tooltip content. If you need to include a link, use a [Popover](/components/popover) instead.

Do not use a Tooltip on a disabled [Button](/components/button) to explain why the button is unavailable. The disabled button isn't focusable. 
```

> Tooltips should contain only information that is useful additional context or guidance for the user. Furthermore, Tooltips are easily overlooked and disappear when not focused on, so consider whether the information may be better served by another component.
> 
> Never put information essential to task completion in a Tooltip. Use [Help Text](/components/help-text), which is persistent and more obvious, for essential information.
> 
> Do not use a Tooltip on a disabled [Button](/components/button) to explain why the button is unavailable. The disabled button isn't focusable. 
> 
> Similarly, do not including an [Anchor](/components/anchor) in Tooltip content because an Anchor isn't focusable. If you need to include a link, use a [Popover](/components/popover) instead.
> 
> Use full sentences and punctuation. Include no more than 3 lines of text. Avoid titles and icons.

Reviewer: ```suggestion
Tooltips can only contain text content, and should contain only information that is useful additional context or guidance for the user. If you need to include an [Anchor](/components/anchor) or icons, use a [Popover](/components/popover) instead.

Never put information essential to task completion in a Tooltip. Furthermore, Tooltips are easily overlooked and disappear when not focused on, so consider whether the information may be better served by another component. Use [Help Text](/components/help-text), which is persistent and more obvious, for essential information.

Do not use a Tooltip on a disabled [Button](/components/button) to explain why the button is unavailable. The disabled button isn't focusable.

Use full sentences and punctuation. Include no more than 3 lines of text.
```

Reorganized and clarified some component guidance to be a bit more accurate. Feel free to update more.

## Do and don't

> <DoDont>

Reviewer: chore: Include the Do and Don't component imports at the top of the file so that the Dos and Don'ts show up on the page.

  <Do
    title="Do"
    body="Use Tooltips to provide additional helpful information or guidance."
    preview={false}
  />
  <Dont title="Don't" body="Don't use Tooltips for information essential to task completion." preview={false} />
</DoDont>

<DoDont>
  <Do title="Do" body="Use Tooltips only on focusable elements." preview={false} />
  <Dont

>     title="Don't use Tooltips on non-focusable elements, like icons and disabled buttons."
>     body="Don't use ."

Reviewer: ```suggestion
    title="Don't"
    body="Don't use Tooltips on non-focusable elements, like icons and disabled buttons."
```

  {StateHookTabsExample}
</LivePreview>

## Composition notes

Tooltips should contain only information that is useful additional context or guidance for the user. Furthermore, Tooltips are easily overlooked and disappear when not focused on, so consider whether the information may be better served by another component.

Never put information essential to task completion in a Tooltip. Use [Help Text](/components/help-text), which is persistent and more obvious, for essential information.

Do not use a Tooltip on a disabled [Button](/components/button) to explain why the button is unavailable. The disabled button isn't focusable. 

Similarly, avoid including an [Anchor](/components/anchor) in Tooltip content because the Anchor isn't focusable. If you need to include a link, use a [Popover](/components/popover) instead.

> Use full sentences and punctuation. Include no more than 3 lines of text. Avoid titles and icons.

Author: done.
